server/routes/allocate-key-workers/controller.ts

```diff
@@ -0,0 +1,47 @@
+import { Request, Response } from 'express'
+import KeyworkerApiService from '../../services/keyworkerApi/keyworkerApiService'
+import { firstNameSpaceLastName } from '../../utils/formatUtils'
+import LocationsInsidePrisonApiService from '../../services/locationsInsidePrisonApi/locationsInsidePrisonApiService'
+
+export class AllocateKeyWorkerController {
+  constructor(
+    private readonly keyworkerApiService: KeyworkerApiService,
+    private readonly locationsApiService: LocationsInsidePrisonApiService,
+  ) {}
+
+  GET = async (req: Request, res: Response): Promise<void> => {
+    const prisonCode = res.locals.user.activeCaseLoad!.caseLoadId!
+    const records = await this.keyworkerApiService.searchPrisoners(req, prisonCode, {
+      query: req.query['query']?.toString() || '',
+      cellLocationPrefix: req.query['location']?.toString() || '',
+    })
+
+    const keyworkers = await this.keyworkerApiService.getKeyworkerMembers(req, prisonCode, { status: 'ALL' })
+    const locations = await this.locationsApiService.getResidentialLocations(req, prisonCode)
+
+    const withoutKeyworker = req.query['withoutKeyworker'] === 'true'
+    res.render('allocate-key-workers/view', {
+      query: req.query['query'],
+      location: req.query['location'],
+      withoutKeyworker,
+      records: records.filter(o => withoutKeyworker || o.keyworker),
+      locations: locations.map(o => o.fullLocationPath),
+      keyworkers: keyworkers.map(o => {
```

Comment:
I've asked in Slack about this. The legacy UI filters out each person's current keyworker. This version will display it as the currently selected option. Either is fine I think once the choice has been made and it can be left till we do the actual allocation ticket

Reply:
Alex wants to follow the behaviour of the existing service meaning the drop down for each person should remove their currently allocated keyworker as an option